routes/api/contacts.js

```diff
@@ -1,80 +1,137 @@
 const express = require('express');
 const router = express.Router();
-const { v4: uuidv4 } = require('uuid');
 const Joi = require('joi');
-const { listContacts, getById, addContact, removeContact, updateContact } = require('../../models/contacts');
-
-// Schematy walidacji
-const contactSchema = Joi.object({
-  name: Joi.string().required(),
-  email: Joi.string().email().required(),
-  phone: Joi.string().required()
-});
-
-const updateContactSchema = Joi.object({
+const {
+  listContacts,
+  getById,
+  addContact,
+  removeContact,
+  updateContact,
+  updateStatusContact,
+} = require('../../models/contacts');
+
+const baseSchema = Joi.object({
   name: Joi.string(),
   email: Joi.string().email(),
-  phone: Joi.string()
-}).or('name', 'email', 'phone');
+  phone: Joi.string(),
+  favorite: Joi.boolean(),
+});
 
-// Trasy
-router.get('/', (req, res) => {
-  const contacts = listContacts();
-  res.json(contacts);
+const validate = (schema, req, res, next) => {
+  const { error } = schema.validate(req.body);
+  if (error) {
+    return res.status(400).json({ message: `Validation error: ${error.details.map(detail => detail.message).join(', ')}` });
+  }
+  next();
+};
+
+const checkRequiredFields = (fields, req, res, next) => {
+  for (const field of fields) {
+    if (!req.body[field]) {
+      return res.status(400).json({ message: `Validation error: missing required field ${field}` });
+    }
+  }
+  next();
+};
+
+router.get('/', async (req, res) => {
+  try {
+    const contacts = await listContacts();
+    res.json(contacts);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
 });
 
-router.get('/:id', (req, res) => {
+router.get('/:id', async (req, res) => {
   const { id } = req.params;
-  const contact = getById(id);
-
-  if (contact) {
-    res.json(contact);
-  } else {
-    res.status(404).json({ message: 'Not found' });
+  try {
+    const contact = await getById(id);
+    if (contact) {
+      res.json(contact);
+    } else {
+      res.status(404).json({ message: 'Not found' });
+    }
+  } catch (error) {
+    res.status(500).json({ message: error.message });
   }
 });
 
-router.post('/', (req, res) => {
-  const { error } = contactSchema.validate(req.body);
-
-  if (error) {
-    return res.status(400).json({ message: `Validation error: ${error.details.map(detail => detail.message).join(', ')}` });
+router.post('/', (req, res, next) => validate(baseSchema, req, res, next), (req, res, next) => checkRequiredFields(['name', 'email', 'phone'], req, res, next), async (req, res) => {
```

Comment:
To się wydaje za bardzo skąplikowane. Lepiej użyć tylko joi wewnątrz funkcji albo jako middleware.